eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1

```diff
@@ -164,59 +176,86 @@ function DeployStressPackage(
         if ($LASTEXITCODE) { return }
     }
 
-    $imageTag = "${registryName}.azurecr.io"
+    $imageTagBase = "${registryName}.azurecr.io"
     if ($repositoryBase) {
-        $imageTag += "/$repositoryBase"
+        $imageTagBase += "/$repositoryBase"
     }
-    $imageTag += "/$($pkg.Namespace)/$($pkg.ReleaseName):${deployId}"
+    $imageTagBase += "/$($pkg.Namespace)/$($pkg.ReleaseName)"
 
-    $dockerFilePath = if ($pkg.Dockerfile) {
-        Join-Path $pkg.Directory $pkg.Dockerfile
-    } else {
-        "$($pkg.Directory)/Dockerfile"
-    }
-    $dockerFilePath = [System.IO.Path]::GetFullPath($dockerFilePath)
+    Write-Host "Creating namespace $($pkg.Namespace) if it does not exist..."
+    kubectl create namespace $pkg.Namespace --dry-run=client -o yaml | kubectl apply -f -
+    if ($LASTEXITCODE) {exit $LASTEXITCODE}
 
-    if ($pushImages -and (Test-Path $dockerFilePath)) {
-        Write-Host "Building and pushing stress test docker image '$imageTag'"
-        $dockerFile = Get-ChildItem $dockerFilePath
-        $dockerBuildFolder = if ($pkg.DockerBuildDir) {
-            Join-Path $pkg.Directory $pkg.DockerBuildDir
+    $dockerFilePaths = @()
+    
+    $genValFile = Join-Path $pkg.Directory "generatedValues.yaml"
+    $genVal = Get-Content $genValFile -Raw | ConvertFrom-Yaml
+    if (Test-Path $genValFile) {
+        $scenarios = $genVal.Scenarios
+        foreach ($scenario in $scenarios) {
+            if ($scenario.image) {
+                $dockerfilePath = Join-Path $pkg.Directory $scenario.image
+                $dockerFilePath = [System.IO.Path]::GetFullPath($dockerFilePath)
+                $dockerFilePaths += $dockerFilePath
+            }
+        }
+    } else {
+        if ($pkg.Dockerfile) {
+            $dockerFilePath = Join-Path $pkg.Directory $pkg.Dockerfile
```

Comment:
Let's print a deprecation warning here? Or do you think we should keep supporting chart.yaml docker configs.

Reply:
I do think we should keep supporting chart.yaml docker configs until we have a default matrix in place

Reply:
I'll still print a message for "we recommend using the matrix because this will soon be deprecated"

Reply:
since we're putting everything inside the matrix

Reply:
In that case should we print a warning message here? The presence of it in chart.yaml won't fail anything otherwise. I'm wondering how to handle migration and config rot.

Reply:
It's been done [here](https://github.com/Azure/azure-sdk-tools/pull/3790/files#diff-cf14bf3e7393a37320cef47678dd2229f9a0385548fadddeae5ff5657f9eb92aR212-R214)